sway-lsp/src/core/token.rs

```diff
@@ -366,9 +366,26 @@ fn handle_expression(exp: Expression, tokens: &mut Vec<Token>) {
             contract_call_params,
             ..
         } => {
-            let ident = method_name.easy_name();
-            let token = Token::from_ident(&ident, TokenType::MethodApplication);
-            tokens.push(token);
+            match &method_name {
+                MethodName::FromType { call_path, .. } => {
+                    if let Some(ident) = call_path.prefixes.get(1) {
+                        // We want to avoid collecting operator methods such as / + * - etc.
+                        if ident.as_str() != "ops" {
+                            let ident = call_path.suffix.clone();
+                            let token = Token::from_span(
+                                ident.span().clone(),
+                                TokenType::MethodApplication,
+                            );
+                            tokens.push(token);
+                        }
+                    }
+                }
+                MethodName::FromModule { method_name, .. } => {
+                    let ident = method_name.clone();
+                    let token = Token::from_ident(&ident, TokenType::MethodApplication);
+                    tokens.push(token);
+                }
+            };
```

Comment:
It looks like the idea behind this PR is to only collect the token in the case that this method call is not one of the methods that are desugared from a short-hand operator like `+`, `/`, `!`, etc. However, it looks like this change might also unintentionally stop collecting all `FromType` method names that don't have at least two prefixes in their call path? I think it might also omit all non-`core` packages that happen to have an `ops` module too?

Maybe we want something like this instead?

```rust
// Check if the given method is a `core::ops` application desugared from short-hand syntax like / + * - etc.
fn desugared_op(method_name: &MethodName) -> bool {
    if let MethodName::FromType { ref call_path, .. } = method_name {
        let prefix0 = call_path.prefixes.get(0).map(|ident| ident.as_str());
        let prefix1 = call_path.prefixes.get(1).map(|ident| ident.as_str());
        if let (Some("core"), Some("ops")) = (prefix0, prefix1) {
            return true;
        }
    }
    false
}

// Don't collect applications of desugared operators due to mismatched ident lengths. 
if !desugared_op(&method_name) {
    let ident = method_name.easy_name();
    let token = Token::from_ident(&ident, TokenType::MethodApplication);
    tokens.push(token);
}
```

Let me know if I'm missing the point :joy:

Reply:
No, that's exactly the point! :) 

Thanks, this solution is definitely the preferred method! Will update this now.